cached_path/schemes/r2.py

```diff
@@ -0,0 +1,72 @@
+"""
+Cloudflare R2.
+"""
+import io
+import os
+from typing import Optional
+
+import boto3.session
+from botocore.config import Config
+
+from .scheme_client import SchemeClient
+from ..common import _split_cloud_path
+
+
+class R2Client(SchemeClient):
+    scheme = "r2"
```

Comment:
Would be good to set the recoverable error types here just like we do with the S3 client.
```suggestion
    recoverable_errors = SchemeClient.recoverable_errors + (
        botocore.exceptions.HTTPClientError,
        botocore.exceptions.ConnectionError,
    )

    scheme = "r2"
```